github-event/handler.go

```diff
@@ -238,3 +268,56 @@ func forward(req []byte, function string, headers map[string]string) (string, in
 
 	return string(body), res.StatusCode, err
 }
+
+func validCustomer(customers []string, owner string) bool {
+	found := false
+	for _, customer := range customers {
+		if len(customer) > 0 &&
+			strings.EqualFold(customer, owner) {
+			found = true
+			break
+		}
+	}
+
+	return found
+}
+
+// getCustomers reads a list of customers separated by new lines
+// who are valid users of OpenFaaS cloud
+func getCustomers(customerURL string) ([]string, error) {
+	customers := []string{}
+
+	if len(customerURL) == 0 {
+		return nil, fmt.Errorf("customerURL was nil")
+	}
+
+	c := http.Client{}
+
+	httpReq, _ := http.NewRequest(http.MethodGet, customerURL, nil)
+	res, reqErr := c.Do(httpReq)
+
+	if reqErr != nil {
+		return customers, reqErr
+	}
+
+	if res.Body != nil {
+		defer res.Body.Close()
+
+		pageBody, _ := ioutil.ReadAll(res.Body)
+		customers = strings.Split(string(pageBody), "\n")
+
+		for i, c := range customers {
+			customers[i] = strings.ToLower(c)
+			customers[i] = strings.TrimSuffix(c, "\r")
```

Comment:
There is something wrong, you are assigning value and then immediately overriding it

Reply:
Yep this was irrational 😄